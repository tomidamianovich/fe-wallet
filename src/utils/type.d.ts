export type CurrencyType = {
  ticker: string
  name: string
  symbol: string
  decimals: number
  type: string
  actions: {
    action: string
    enabled: boolean
  }[]
  url_images: {
    image_png: string
    image_svg: string
  }
  color: string | null
  categories: string[]
  is_favorite: boolean
}

export type RateType = {
  ticker: string
  buy_rate: string
  sell_rate: string
  variation: string
}

export type BalanceType = {
  ticker: string
  amount: number
  account_limit: number
}

export type TransactionActionType =
  | 'DEPOSIT'
  | 'WITHDRAWAL'
  | 'SEND'
  | 'RECEIVE'
  | 'BUY'
  | 'SWAP'
export type TransactionStatusType = 'PENDING' | 'FAILURE' | 'SUCCESS'

export type TransactionType = {
  date: Date
  action: TransactionActionType
  ticker: string
  via: string
  status: TransactionStatusType
  comision: number
  amount: number
  total: number
}

export interface transaction {
  ['DEPOSIT']: TransactionActionType
  ['WITHDRAWAL']: TransactionActionType
  ['SEND']: TransactionActionType
  ['RECEIVE']: TransactionActionType
  ['BUY']: TransactionActionType
  ['SWAP']: TransactionActionType
}
export interface transactionTitle {
  DEPOSIT: string
  WITHDRAWAL: string
  SEND: string
  RECEIVE: string
  BUY: string
  SWAP: string
}

export interface transactionStatus {
  ['PENDING']: TransactionStatusType
  ['FAILURE']: TransactionStatusType
  ['SUCCESS']: TransactionStatusType
}

export type Link = {
  name: string
  to: string
  icon: string
}

export type EventSelect = {
  target: {
    value: TransactionActionType
  }
}

export type NavLinkType = link

export type CurrencyState = CurrencyType[]
export type RateState = RateType[]
export type NavLinkState = NavLinkType[]
export type BalanceState = BalanceType[]
export type TransactionState = TransactionType[]

export type CurrencyAction = {
  type: strings
  payload: CurrencyState
}

export type RateAction = {
  type: string
  payload: RateState
}

export type NavLinkAction = {
  type: string
  payload: NavLinkState
}

export type BalanceAction = {
  type: string
  payload: BalanceState
}

export type CurrentBalanceAction = {
  type: string
  payload: BalanceType
}

export type TransactionsAction = {
  type: string
  payload: TransactionState
}

export type BalanceWithCurrencyInfo = BalanceType &
  CurrencyType & { sell_rate: string }

export type CurrencyDispatchType = (args: CurrencyAction) => CurrencyAction
export type RateDispatchType = (args: RateAction) => RateAction
export type NavLinkDispatchType = (args: RateAction) => NavLinkAction
export type BalanceDispatchType = (args: BalanceAction) => BalanceAction
export type TransactionDispatchType = (
  args: TransactionsAction,
) => TransactionsAction

// Combined Types

export type CombinedState = {
  CurrencyReducer: CurrencyState
  RateReducer: RateState
  NavLinkReducer: NavLinkState
  BalanceReducer: BalanceState
  CurrentBalanceReducer: BalanceType
  TransactionsReducer: TransactionState
}

export type CombinedAction = {
  CurrencyReducer: CurrencyAction
  RateReducer: RateAction
  NavLinkReducer: NavLinkAction
  BalanceReducer: BalanceAction
  CurrentBalanceReducer: CurrentBalanceAction
  TransactionsReducer: TransactionsAction
}

export type CombinedDispatchType = (args: CombinedAction) => CombinedAction
