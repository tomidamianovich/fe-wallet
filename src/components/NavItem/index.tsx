import React from 'react';
import './styles/index.scss';
import { NavLink } from "react-router-dom";
import Icon from "../Icon";

type Props = {
  to: string;
  name: string;
  icon: string;
};

const NavItem: React.FC<Props> = ({to, name, icon}) => 
  <li className='fe-wallet__content__navigation__list__item'>
    <NavLink 
      to={to}
      className={({ isActive }) => `fe-wallet__content__navigation__list__item__link ${isActive && 'fe-wallet__content__navigation__list__item__link--active'}`}
    >
      <div className='fe-wallet__content__navigation__list__item__link__content'>
        <Icon icon={icon}/>
        <span>
          {name}
        </span>
      </div>
    </NavLink>
  </li>;
;

export default NavItem;