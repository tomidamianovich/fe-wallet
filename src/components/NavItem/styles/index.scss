.fe-wallet__content__navigation__list {
  &__item {
    height: 60px;
    display: flex;
    align-items: center;
    justify-content: left;
    margin-top: 12px;

    &__link {
      color: black;
      text-decoration: none;
      font-size: 16px;
      line-height: 16px;
      letter-spacing: 0px;
      text-align: left;
      height: 48px;
      width: 100%;
      display: flex;
      align-items: center;

      &__content {
        display: flex;
        height: 60px;
        align-items: center;
      }

      .icon {
        height: 14.12px;
        width: 16px;
        left: 0px;
        top: 1.066680908203125px;
        border-radius: 0px;
        margin-left: 21px;
        color: red;
        margin-right: 12px;
        path {
          fill: black;
        }
      }

      &--active {
        font-weight: 700;  
        border-left: 3px solid;
        border-color:#7908FF;

        .icon {
          height: 14.12px;
          width: 16px;
          left: 0px;
          top: 1.066680908203125px;
          border-radius: 0px;
          margin-left: 21px;
          color: red;
          margin-right: 12px;
          path {
            fill: #7908FF;
          }
        }
      }
    }
  }
}