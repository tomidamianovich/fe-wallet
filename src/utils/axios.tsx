import { RateType, CurrencyType } from '../utils/type';
const axios = require('axios').default;

type responseType = {
  data: RateType[] | CurrencyType[]
}
  
const requestHandler = async (url:string): Promise<any> => 
  new Promise((resolve, reject) => 
    axios.get(url)
    .then((response: responseType) => resolve(response?.data))
    .catch((err:any) => reject(err))
  );

export {
  requestHandler,
}