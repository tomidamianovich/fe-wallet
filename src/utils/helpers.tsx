const handleFormatValues = (value:number) => value.toFixed(2);

export {
  handleFormatValues
}