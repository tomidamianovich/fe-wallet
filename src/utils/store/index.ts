import currencies from './currencies';
import cryptoCurrencies from './cryptoCurrencies';
import fiatCurrencies from './fiatCurrencies';
import balances from './balances';
import rates from './rates';

export {
  currencies,
  fiatCurrencies,
  cryptoCurrencies,
  rates,
  balances
}