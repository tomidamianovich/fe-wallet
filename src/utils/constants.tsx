import { transaction, transactionTitle, transactionStatus } from '../utils/type';

const ENDPOINTS = {
  CURRENCIES: 'http://localhost:3001/currencies',
  RATES: 'http://localhost:3001/rates',
  BALANCES: 'http://localhost:3001/balance'
}

const PANEL = {
  HEADING: "Mi billetera",
  SUBHEADING: "En tu billetera vas a poder almacenar todas las criptomonedas que compres en ripio.",
}

const TRANSACTIONS:{
  HEADING: string,
  HEADER: string[],
  TYPES: transaction,
  TYPES_TITLES: transactionTitle,
  STATUS: transactionStatus
} = {
  HEADING: "Mis movimientos",
  HEADER: [
    "Fecha",
    "Tipo de Operación",
    "Método",
    "Estado",
    "Comisión",
    "Monto",
  ],
  TYPES: {
    DEPOSIT: "DEPOSIT",
    WITHDRAWAL: "WITHDRAWAL",
    SEND: "SEND",
    RECEIVE: "RECEIVE",
    BUY: "BUY",
    SWAP: "SWAP"
  },
  TYPES_TITLES: {
    DEPOSIT: "Ingreso",
    WITHDRAWAL: "Retiro",
    SEND: "Envio",
    RECEIVE: "Recibo",
    BUY: "Compra",
    SWAP: "Swap"
  },
  STATUS: {
    PENDING: "PENDING",
    FAILURE: "FAILURE",
    SUCCESS: "SUCCESS"
  }
}

const DEFAULT_BALANCE = {
  ticker: "",
  amount: 0,
  account_limit: 0,
  name: "",
  symbol: "",
  decimals: 2,
  type: "",
  actions: [{
    action: "",
    enabled: true
  }],
  url_images: {
    image_png: "",
    image_svg: "",
  },
  color: "",
  categories: [],
  is_favorite: true,
  sell_rate: ""
}

const COMISIONS = {
  BUY: 0.003,
  DEPOSIT: 0,
  WITHDRAWAL: 0.0004,
  SEND: 0,
  RECEIVE: 0,
  SWAP: 0.00004
}

type Link = {
  name: string;
  to: string;
  icon: string;
};

const NAV_LINKS:Link[] = [
  {
    name: 'Panel',
    to: 'panel',
    icon: 'panel'
  },
  {
    name: 'Billetera',
    to: 'wallet',
    icon: 'wallet'
  },
  {
    name: 'Cotizaciones',
    to: 'market-rates',
    icon: 'market-rates'
  },
  {
    name: 'Créditos',
    to: 'credit',
    icon: 'credit'
  },
  {
    name: 'Exchange',
    to: 'exchange',
    icon: 'exchange'
  },
  {
    name: 'Launchpad',
    to: 'launchpad',
    icon: 'launchpad'
  }
];

const WORDINGS = {
  BALANCE: {
    DETAIL: {
      INPUT: {
        LABELS: {
          TRANSACTION: "Seleccione el tipo de operacion a realizar",
          AMOUNT: "Ingrese cantidad",
          ARIA_LABEL : "Monto de la transacción"
        },
        MESSAGES: {
          INSUFFICIENT_AMOUNT: "Dinero insuficiente",
          AMOUNT_TO_DEBIT: "Monto a debitar",
          AMOUNT_NEEDED: "Monto necesario",
          ACTUAL_BALANCE: "Monto actual"
        }
      },
      ACTIONS: {
        CANCEL: {
          LABEL: "Cancelar",
          ARIA_LABEL: "Cancelar transacción",
        },
        CONFIRM: {
          LABEL: "Confirmar",
          ARIA_LABEL: "Confirmar transacción",
        }
      },
      CARD: {
        ACCOUNT: "Cuenta"
      }
    },
    SUMMARY: {
      DROPDOWN: {
        ARIA_LABEL: "Seleccionar moneda para ver balance"
      }
    }
  },
  CONVERTER: {
    FROM: {
      DROPDOWN: {
        ARIA_LABEL: "Seleccionar crypto origen",
        NAME: "from"
      },
      INPUT: {
        ARIA_LABEL: "Cripto a convertir",
        NAME: "from_amount"
      }
    },
    TO: {
      DROPDOWN: {
        ARIA_LABEL: "Seleccionar crypto destino",
        NAME: "to"
      },
      INPUT: {
        ARIA_LABEL: "Cripto convertida",
        NAME: "to_amount"
      }
    },
    BUTTON: {
      ARIA_LABEL: "Convertir",
      LABEL: "Convertir",
    }
  },
  CURRENCIES: {
    TYPES: {
      FIAT: "FIAT",
      CRYPTO: "CRYPTO"
    }
  },
  MAINTENANCE: {
    TITLE: "Pagina en Mantenimiento",
    SUBTITLE: {
      BEGINNING: "Oops! Estamos trabajando en la pagina ",
      ENDING: ". Ingrese nuevamente a esta seccion más tarde."
    }
  },
  NAVIGATION: {
    NAV_ITEM: {
      NAME: "Salir"
    }
  },
  NOT_FOUND: "Pagina no encontrada.",
  ERROR_FOUND: "Oops. Ocurrio un error",
  PAGES: {
    WALLET: "wallet",
    CREDIT: "credit",
    EXCHANGE: "exchange",
    LAUNCHPAD: "launchpad",
    MARKET_RATES: "market-rates",
    PANEL: "panel",
  }
}

export {
  ENDPOINTS,
  PANEL,
  TRANSACTIONS,
  DEFAULT_BALANCE,
  COMISIONS,
  NAV_LINKS,
  WORDINGS
}