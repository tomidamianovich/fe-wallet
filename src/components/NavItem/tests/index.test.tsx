import { render } from '@testing-library/react'
import NavItem from '../'
import { NAV_LINKS } from '../../../utils/constants'
import { BrowserRouter } from 'react-router-dom'

test('renders NavItem main page', () => {
  const link = NAV_LINKS[0]
  const { asFragment } = render(
    <BrowserRouter>
      <NavItem {...link} />
    </BrowserRouter>,
  )
  expect(asFragment()).toMatchSnapshot()
})
