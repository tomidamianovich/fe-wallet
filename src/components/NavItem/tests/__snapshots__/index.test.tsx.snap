// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NavItem main page 1`] = `
<DocumentFragment>
  <li
    class="fe-wallet__content__navigation__list__item"
  >
    <a
      class="fe-wallet__content__navigation__list__item__link false"
      href="/fe-wallet"
    >
      <div
        class="fe-wallet__content__navigation__list__item__link__content"
      >
        <svg
          class="icon"
          height="16"
          viewBox="0 0 15 16"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0.471477 7.5952H1.88308V13.7121C1.88308 13.972 2.09374 14.1827 2.35361 14.1827H5.1768C5.43667 14.1827 5.64734 13.972 5.64734 13.7121V9.7126C5.64734 9.06293 6.174 8.53627 6.82367 8.53627C7.47334 8.53627 8 9.06293 8 9.7126V13.7121C8 13.972 8.21066 14.1827 8.47053 14.1827H13.6464C13.9063 14.1827 14.1169 13.972 14.1169 13.7121V7.5952H15.5285C15.963 7.5952 16.1653 7.05666 15.8384 6.77056L8.30985 0.183102C8.13245 0.027874 7.86756 0.027874 7.69015 0.183102L0.161629 6.77056C-0.165345 7.05666 0.0370036 7.5952 0.471477 7.5952ZM8.00007 1.16244L14.2763 6.65413H13.6465C13.3866 6.65413 13.1759 6.8648 13.1759 7.12467V13.2416H8.94113V9.7126C8.94113 8.54319 7.99314 7.5952 6.82374 7.5952C5.65433 7.5952 4.70634 8.54319 4.70634 9.7126V13.2416H2.82421V7.12467C2.82421 6.8648 2.61354 6.65413 2.35368 6.65413H1.72384L8.00007 1.16244Z"
            fill-rule="evenodd"
          />
        </svg>
        <span>
          Panel
        </span>
      </div>
    </a>
  </li>
</DocumentFragment>
`;
